novawallet/Common/Services/DelegatedAccounts/Repositories/MultisigAccountsRepository.swift

```diff
@@ -0,0 +1,92 @@
+import Foundation
+import SubstrateSdk
+import Operation_iOS
+
+final class MultisigAccountsRepository {
+    private let chain: ChainModel
+
+    private let mutex = NSLock()
+
+    private var multisigsBySignatories: [AccountId: [DiscoveredMultisig]] = [:]
+
+    init(chain: ChainModel) {
+        self.chain = chain
+    }
+
+    private func mapToDelegatedAccounts(_ multisigs: [DiscoveredMultisig]) -> [DelegatedAccount] {
+        multisigs.map { DelegatedAccount.multisig($0) }
+    }
+}
+
+// MARK: DelegatedAccountsRepositoryProtocol
+
+extension MultisigAccountsRepository: DelegatedAccountsRepositoryProtocol {
+    func fetchDelegatedAccountsWrapper(
+        for delegators: Set<AccountId>
+    ) -> CompoundOperationWrapper<[AccountId: [DelegatedAccount]]> {
+        let cachedMultisigsForSignatories = delegators
+            .map { (signatory: $0, multisigs: multisigsBySignatories[$0]) }
+            .reduce(into: [:]) { $0[$1.signatory] = $1.multisigs }
+
+        let cachedSignatories = Set(cachedMultisigsForSignatories.keys)
+        let nonCachedSignatories = delegators.subtracting(cachedSignatories)
+
+        guard !nonCachedSignatories.isEmpty else {
+            return .createWithResult(
+                cachedMultisigsForSignatories.mapValues(mapToDelegatedAccounts)
+            )
+        }
+
+        guard let apiURL = chain.externalApis?.getApis(for: .multisig)?.first?.url else {
+            return .createWithResult([:])
+        }
+
+        let fetchFactory = SubqueryMultisigsOperationFactory(
+            url: apiURL
+        )
+
+        let fetchOperation = fetchFactory.createDiscoverMultisigsOperation(for: nonCachedSignatories)
+
+        let mapOperation = ClosureOperation<[AccountId: [DelegatedAccount]]> { [weak self] in
+            guard let self else {
```

Comment:
I don't like such implicit assumptions that operation factory should stay alive till operation completes. Better to catch strongly i think